book/content/code_styling/Code_Styling.md

```diff
@@ -0,0 +1,137 @@
+# Writing Clear and (Human) Readable Code
+## Introduction
+Have you ever opened a syntax or script file two years after running an analysis, only to find that not only have you no memory of the code, but you've forgotten the language too?
+On top of this you have a load of comments, such as "# Add this later", or "# WHAT WAS I THINKING???", and variables of "df_1_new", "new_dat_4_r".  If you have a larger project, perhaps your filenames include "dat_analsis_1_FINAL_.R" and "data_analysis_2_FINAL_2.R"?
+
+This is frustrating enough when it is just you reading back on it, but imagine cloning these files from github, or the Open Science Framework
+Not only do you have no memory of what was meant, but you never knew in the first place!
+
+This chapter aims to introduce some principals of code hygine, including style, naming conventions, creating useful comments, and will briefly discuss the concept of *linting*, which is a real time error checking process that many integrated development environments (IDE) include.  Keeping this advice in mind will help you create reusable, and easily adaptable code.
+
+## Code Styling
+
+Style guidelines differ between organisations, languages, and over time ^[The Python style guide (PEP) is now in its 8th edition.].
+It is important that you work to a framework that is best for your purposes, bearing in mind that consistency makes for easier reading and understanding.
+
+Style guidelines include advice for file naming, variable naming, comments, and the use of whitespace and brackets.
+For examples of style guides, follow the following links:
+
+* [PEP8](https://www.python.org/dev/peps/pep-0008/) for Python
+* [Hadley Wickham's](http://adv-r.had.co.nz/Style.html) style guide for R
+* [Google's](https://google.github.io/styleguide/Rguide.xml) style guide for R
+* [Microsoft's](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions) style guide for C#.
+
+## File Naming
+
+The Centre for Open Science has some useful suggestions for naming of files, particularly ensuring that they readable both to humans and computers.  This includes avoiding the use of wildcard characters (@£$%) and using underscores ("_") to delimit information, and dashes ("-") to conjunct information or spaces.  They also suggest dating or numbering files, and avoiding words like FINAL (or FINAL-FINAL).  The dating suggestion is 
```

Comment:
the _ here is accidentally enabling the markdown _italics_ `_italics_` - we might want to try escaping it if possible, using a slash, e.g. `\_` :)

Reply:
Thanks Yo!!